CMakeLists.txt

```diff
@@ -83,7 +83,11 @@ set(WITH_SDL CACHE STRING "The SDL version to use: 1 or 2 or empty (use any)")
 set(WITH_QT CACHE STRING "The Qt version to use: 4 or 5 or empty (use any)")
 
 # Build types
-set(default_UNITY_BUILD ON)
+if(MACOSX)
+	set(default_UNITY_BUILD OFF)
```

Comment:
You should exclude the .mm files from the unity build like we already do for .rc files on Windows instead of disabling it completely. Also, you are only changing the default here.